pinot-core/src/main/java/org/apache/pinot/core/accounting/CPUMemThreadLevelAccountingObjects.java

```diff
@@ -32,47 +30,49 @@
  */
 public class CPUMemThreadLevelAccountingObjects {
 
-  public static class StatsDigest {
-
-    // The current usage sampling for each thread
-    final long[] _currentStatsSample;
-    // The previous usage sampling for each thread
-    final long[] _lastStatSample;
-    // The aggregated usage sampling for the finished tasks of a (still) running queries
-    final HashMap<String, Long> _finishedTaskStatAggregator;
-
-    StatsDigest(int numThreads) {
-      _currentStatsSample = new long[numThreads];
-      _lastStatSample = new long[numThreads];
-      _finishedTaskStatAggregator = new HashMap<>();
-    }
-  }
-
   /**
-   * Entry to track the task execution status of a worker/runner given thread
+   * Entry to track the task execution status and usage stats of a Thread
    */
-  public static class TaskEntryHolder {
-    AtomicReference<TaskEntry> _threadTaskStatus = new AtomicReference<>(null);
+  public static class ThreadEntry {
+    // current query_id, task_id of the thread; this field is accessed by the thread itself and the accountant
+    AtomicReference<TaskEntry> _currentThreadTaskStatus = new AtomicReference<>(null);
+    // current sample of thread memory usage/cputime ; this field is accessed by the thread itself and the accountant
+    volatile long _currentThreadCPUTimeSampleMS = 0;
+    volatile long _currentThreadMemoryAllocationSampleBytes = 0;
+
+    // previous query_id, task_id of the thread, this field should only be accessed by the accountant
+    TaskEntry _previousThreadTaskStatus = null;
+    // previous query_id, task_id of the thread, this field should only be accessed by the accountant
```

Comment:
I think this comment seems incorrect / copy of the previous one ?

Reply:
fixed